### PR TITLE
Revise details for sidecar containers

### LINKED ISSUE #44334: Redundant information present in sidecar containers page
**This is a Bug Report**
**Problem:**
1. Below text is no longer needed as sidecar containers are now entered into beta state and they are enable by default.

> By default, this feature is not available in Kubernetes. To avail this feature, you need to enable the [feature gate](https://v1-28.docs.kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) named SidecarContainers.

2. The reference link of sidecar containers present in [featurgates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates) page is wrongly redirecting to [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) page.

**Proposed Solution:**
1. Remove the text below from sidecar containers documentation.
> By default, this feature is not available in Kubernetes. To avail this feature, you need to enable the [feature gate](https://v1-28.docs.kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) named SidecarContainers.

2. Change reference link present in [featuregates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates) page from [init-containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) page to [sidecar containers](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/) page.

**Page to Update:**
https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates
https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/

### PR DESCRIPTION
**Pull request by T-Lakshmi**
Fixes #44334

### CODE REVIEW

**Comment by T-Lakshmi** [comment]
**Preview pages:**
https://deploy-preview-44335--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/pods/sidecar-containers/

https://deploy-preview-44335--kubernetes-io-main-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/#feature-gates

File: content/en/docs/concepts/workloads/pods/sidecar-containers.md [hunk 1]
```diff
@@ -57,9 +57,6 @@ has finished.
 Here's an example of a Job with two containers, one of which is a sidecar:
 
 {{% code_sample language="yaml" file="application/job/job-sidecar.yaml" %}}
-By default, this feature is not available in Kubernetes. To avail this feature, you
-need to enable the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
-named `SidecarContainers`.
```
Comment (reylejano) [inline content/en/docs/concepts/workloads/pods/sidecar-containers.md:62]: This change needs to be verified as the related issue states that beta APIs are on by default but as of https://github.com/kubernetes/enhancements/issues/3136 , beta APIs are off by default

**Comment by T-Lakshmi** [comment]
/remove language zh

Removing the label which was added due to merge conflict.

File: content/en/docs/concepts/workloads/pods/sidecar-containers.md (hunk 1 above)
Comment (skaegi) [reply content/en/docs/concepts/workloads/pods/sidecar-containers.md:62]: L20-21 should also be update to reflect what @matthyx had suggested in https://github.com/matthyx/website/commit/29dff23d39e4b855cd32ba6db98cc703645108af

It looks like this change originally in `init-containers.md` was lost when `sidecar-containers.md` was added. It is important as at first glance the current wording gives the impression that sidecars containers are still alpha and need a feature flag set.

```
(Before)
Starting with Kubernetes 1.28 in alpha, a feature gate named `SidecarContainers`...

(After)
Enabled by default with Kubernetes 1.29, a feature gate named `SidecarContainers`
````

Comment (matthyx) [reply content/en/docs/concepts/workloads/pods/sidecar-containers.md:62]: @skaegi it is a good catch, we're indeed in beta
@reylejano the feature is ON by default:
https://github.com/kubernetes/kubernetes/blob/v1.29.0/pkg/features/kube_features.go#L1195

Comment (T-Lakshmi) [reply content/en/docs/concepts/workloads/pods/sidecar-containers.md:62]: @skaegi, @matthyx,
Thanks for the inputs.
I have updated the PR with suggestions https://github.com/matthyx/website/commit/29dff23d39e4b855cd32ba6db98cc703645108af.

**Comment by reylejano** [comment]
Thank you for addressing comments
/lgtm

**Comment by T-Lakshmi** [comment]
@reylejano Thanks for reviewing the PR.

/assign @tengqm
@tengqm, Requesting for PR approval, PTAL!
Thanks.

File: content/en/docs/concepts/workloads/pods/sidecar-containers.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ security, or data synchronization, without directly altering the primary applica
 
 ## Enabling sidecar containers
 
-Starting with Kubernetes 1.28, a
+Enabled by default with Kubernetes 1.29, a
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/sidecar-containers.md:20]: nit: Can you reword to the sentence:
Starting with Kubernetes v1.29, the SidecarContainers [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled by default.
SidecarContainers allows you to specify a `restartPolicy` for containers listed ...

**Pull request merged**